mypyc/genops.py

```diff
@@ -925,6 +994,13 @@ def __init__(self,
         self.callable_class_names = set()  # type: Set[str]
         self.options = options
 
+        # These variables keep track of values that are needed only when constructing a
```

Comment:
Try pulling all of these out into a class where they are non-optional and have IRBuilder just hang on to that class. That means just one thing to check.